### DISCUSSION TITLE
iRAF error on Mac m1 pro

### DISCUSSION THREAD

**Discussion by tudes0** [iraf-community/iraf#169] [Q&A] [answered]
Hi , I have problem with running IRAF. I have installed everything as described, everything works well ;
however when I put  "cl " command and try to open the IRAF, the system gives "Bad CPU executable" error. Any help will be appreciated. Thank you. ( my computer: MacBook Pro Big Sur )

"(iraf27) Bilals-MBP:~ tudes$ cl
/Users/tudes/Documents/opt/anaconda3/envs/iraf27/bin/cl: line 40: /Users/tudes/Documents/opt/anaconda3/envs/iraf27/iraf//bin.macosx/ecl.e: Bad CPU type in executable"

**Comment by olebole** [comment, marked as answer]
You probably used the Astroconda IRAF distribution, right? This doesn't work on Mac M1 computers, since it is 32 bit and 32 bit are not supported anymore.

You need to compile IRAF yourself. Since the M1 support is not released yet, you must clone [the git repository](https://github.com/iraf-community/iraf) and then proceed as described in the [INSTALL.md](https://github.com/iraf-community/iraf/blob/main/INSTALL.md) document.
